oat-sa/tao-core
# Feature/AUT-3636/Apply the Solar Design to the Test Authoring

### Related to: https://oat-sa.atlassian.net/browse/AUT-3636

Requires: 
 - [ ] https://github.com/oat-sa/extension-tao-testqti/pull/2473

### Summary

Apply the Solar Design to the test authoring

![image](https://github.com/oat-sa/tao-core/assets/1500098/485d662d-094d-4f08-b4ed-29f6ce3733ae)

### Details

To activate the feature, you need the feature flag `FEATURE_FLAG_SOLAR_DESIGN_ENABLED`.
```sh
php index.php 'oat\tao\scripts\tools\FeatureFlag\FeatureFlagTool' -s FEATURE_FLAG_SOLAR_DESIGN_ENABLED -v true
```

Once activated, the CSS class `solar-design` is added to the body element of the page. Then, the stylesheet automatically applies.

The Solar Design has been partially replicated in the CSS variables. For compatibility reasons, some have been simplified.

The test authoring is now following the Solar Design.
A few discrepancies were also addressed:
- the feedbacks
- some icons
- a few form fields

### How to test

- check out the branch: `git checkout -t origin/feature/AUT-3636/solar-design-authoring`
- activate the feature flag: `php index.php 'oat\tao\scripts\tools\FeatureFlag\FeatureFlagTool' -s FEATURE_FLAG_SOLAR_DESIGN_ENABLED -v true`
- log in to TAO backoffice
- look at the test authoring, compare with the UX mockups
